libpince/typedefs.py

```diff
@@ -333,31 +333,42 @@ class SCAN_TYPE:
     @staticmethod
     def get_list(scan_mode, value_type):
         if scan_mode == SCAN_MODE.NEW:
-            list = [
-                SCAN_TYPE.EXACT,
-                SCAN_TYPE.NOT,
-                SCAN_TYPE.LESS,
-                SCAN_TYPE.MORE,
-                SCAN_TYPE.BETWEEN,
-                SCAN_TYPE.UNKNOWN,
-            ]
+            if value_type != SCAN_INDEX.AOB and value_type != SCAN_INDEX.STRING:
+                list = [
+                    SCAN_TYPE.EXACT,
+                    SCAN_TYPE.NOT,
+                    SCAN_TYPE.LESS,
+                    SCAN_TYPE.MORE,
+                    SCAN_TYPE.BETWEEN,
+                    SCAN_TYPE.UNKNOWN,
+                ]
+            else:
+                list = [
+                    SCAN_TYPE.EXACT,
+                    SCAN_TYPE.UNKNOWN,
+                ]
+
         else:
-            list = [
-                SCAN_TYPE.EXACT,
-                SCAN_TYPE.NOT,
-                SCAN_TYPE.INCREASED,
-                SCAN_TYPE.INCREASED_BY,
-                SCAN_TYPE.DECREASED,
-                SCAN_TYPE.DECREASED_BY,
-                SCAN_TYPE.LESS,
-                SCAN_TYPE.MORE,
-                SCAN_TYPE.BETWEEN,
-                SCAN_TYPE.CHANGED,
-                SCAN_TYPE.UNCHANGED,
-            ]
-
-        if value_type == SCAN_INDEX.AOB or value_type == SCAN_INDEX.STRING:
-            del list[1]
+            if value_type != SCAN_INDEX.AOB and value_type != SCAN_INDEX.STRING:
+                list = [
+                    SCAN_TYPE.EXACT,
+                    SCAN_TYPE.NOT,
+                    SCAN_TYPE.INCREASED,
+                    SCAN_TYPE.INCREASED_BY,
+                    SCAN_TYPE.DECREASED,
+                    SCAN_TYPE.DECREASED_BY,
+                    SCAN_TYPE.LESS,
+                    SCAN_TYPE.MORE,
+                    SCAN_TYPE.BETWEEN,
+                    SCAN_TYPE.CHANGED,
+                    SCAN_TYPE.UNCHANGED,
+                ]
+            else:
+                list = [
+                    SCAN_TYPE.EXACT,
+                    SCAN_TYPE.CHANGED,
```

Comment:
Strings and AOB support only `EXACT` in this scenario, not `CHANGED` or `UNCHANGED`

Reply:
Also please make sure you use the Black formatter. Your previous PR was unformatted.

Reply:
> Strings and AOB support only EXACT in this scenario, not CHANGED or UNCHANGED

Missed that.

> Also please make sure you use the Black formatter. Your previous PR was unformatted.

Could you explain what is meant by black formatter? I did read in `CONTRIBUTING.md` but didn't understand.

Reply:
We're using this formatter for PINCE: https://github.com/psf/black

You have to use that formatter before any PR to make sure that the code style is consistent.

Reply:
Fixed it. But regarding black formatter, it showed everything was fine and nothing changed.

Reply:
This is still wrong.

`STRING` does not accept `CHANGED` or `UNCHANGED` like I've said before.
Removing `UNKNOWN` from `AOB` was also unnecessary.

Revert your last commit and just remove the `CHANGED` and `UNCHANGED` value from **BOTH** `STRING` and `AOB` like previously stated.

Reply:
Understood